File: backend/internal/model/merchant.go
package model

import "time"

type Merchant struct {
	ID              int64           `json:"id" gorm:"primaryKey;autoIncrement"`
	UserID          int64           `json:"user_id" gorm:"not null;index"`
	ShopName        string          `json:"shop_name" gorm:"type:varchar(200);not null"`
	Industry        string          `json:"industry" gorm:"type:varchar(50);not null;index"` // catering/retail/fresh/bakery
	LogoURL         string          `json:"logo_url" gorm:"type:varchar(500)"`
	Address         string          `json:"address" gorm:"type:text"`
	Latitude        float64         `json:"latitude" gorm:"type:decimal(10,7)"`
	Longitude       float64         `json:"longitude" gorm:"type:decimal(10,7)"`
	ContactPhone    string          `json:"contact_phone" gorm:"type:varchar(20)"`
	BusinessHours   string          `json:"business_hours" gorm:"type:varchar(100)"`
	Announcement    string          `json:"announcement" gorm:"type:text"`
	LicenseURL      string          `json:"license_url" gorm:"type:varchar(500)"`
	IDCardURL       string          `json:"id_card_url" gorm:"type:varchar(500)"`
	ShopPhotos      JSONStringSlice `json:"shop_photos" gorm:"type:jsonb"`
	AuditStatus     string          `json:"audit_status" gorm:"type:varchar(20);not null;default:pending"` // pending/approved/rejected
	AuditRemark     string          `json:"audit_remark" gorm:"type:text"`
	VersionPlan     string          `json:"version_plan" gorm:"type:varchar(20);not null;default:basic"` // basic/pro/chain
	PlanExpireAt    *time.Time      `json:"plan_expire_at"`
	MiniappAppid    string          `json:"miniapp_appid" gorm:"type:varchar(100)"`
	MiniappStatus   string          `json:"miniapp_status" gorm:"type:varchar(20);default:pending"` // pending/uploading/auditing/published/rejected
	Status          string          `json:"status" gorm:"type:varchar(20);not null;default:active;index"` // active/frozen
	CreatedAt       time.Time       `json:"created_at"`
	UpdatedAt       time.Time       `json:"updated_at"`

	// 关联
	User User `json:"user,omitempty" gorm:"foreignKey:UserID"`
}

func (Merchant) TableName() string { return "merchants" }

File: frontend/admin/src/api/merchant.ts
import request from './request'
import type { PaginatedData } from './request'

export interface MerchantListParams {
  page?: number
  page_size?: number
  keyword?: string
  industry?: string
  audit_status?: string
  status?: string
}

export interface Merchant {
  id: number
  shop_name: string
  industry: string
  logo_url: string
  address: string
  contact_phone: string
  audit_status: 'pending' | 'approved' | 'rejected'
  version_plan: 'basic' | 'pro' | 'chain'
  plan_expire_at: string
  status: 'active' | 'frozen'
  created_at: string
  user_id: number
}

export interface MerchantDetail extends Merchant {
  license_url: string
  id_card_url: string
  shop_photos: string[]
  audit_remark: string
  miniapp_status: string
}

export interface AuditParams {
  audit_status: 'approved' | 'rejected'
  audit_remark?: string
}

export interface DashboardStats {
  total_merchants: number
  active_merchants: number
  pending_audit: number
  total_orders_today: number
  total_gmv_today: number
  merchant_growth: number
  order_growth: number
}

export const merchantApi = {
  list: (params: MerchantListParams) =>
    request.get<unknown, PaginatedData<Merchant>>('/admin/merchants', { params }),

  detail: (id: number) =>
    request.get<unknown, MerchantDetail>(`/admin/merchants/${id}`),

  audit: (id: number, params: AuditParams) =>
    request.put(`/admin/merchants/${id}/audit`, params),

  freeze: (id: number, frozen: boolean) =>
    request.put(`/admin/merchants/${id}/freeze`, { frozen }),

  stats: (id: number) =>
    request.get(`/admin/merchants/${id}/stats`),

  dashboard: () =>
    request.get<unknown, DashboardStats>('/admin/dashboard')
}
